client/client.go

```diff
@@ -247,7 +249,10 @@ func (c *Client) downloadTimestamp() error {
 	raw, err := c.remote.GetMeta(role, maxSize)
 	var s *data.Signed
 	if err != nil || len(raw) == 0 {
-		if err, ok := err.(*store.ErrMetaNotFound); ok {
+		if err, ok := err.(store.ErrMetaNotFound); ok {
+			return err
+		}
+		if old == nil {
 			return err
```

Comment:
But `if err != nil || len(raw) == 0` implies we're handling cases where `len(raw) == 0` and `err == nil`. In this case, the function will return nil (no error). Is it really correct us to return nil on an empty file, but ErrMetaNotFound on a 404?